source_ds9/mm_flusher.c

```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: ISC
+//
+// Copyright (c) 2008, Mukunda Johnson (mukunda@maxmod.org)
+// Copyright (c) 2023, Lorenzooone (lollo.lollo.rbiz@gmail.com)
+
+#include "mm_flusher.h"
+#include "multiplatform_defs.h"
+#include "useful_qualifiers.h"
+#include "mm_main9.h"
+#include <stddef.h>
+
+ARM_TARGET void mmFlushBank() {
+    uintptr_t bank_addr = (uintptr_t)mmMemoryBank;
+    for(size_t i = 0; i < ((mmModuleCount + mmSampleCount + 32 - 1 + (bank_addr & 31)) / 32); i++)
+        CP15_CleanAndFlushDCacheEntry((bank_addr & (~31)) + (i * 32));
```

Comment:
If you rely on libnds anyway, can't you remove "useful_qualifiers.h"?

Reply:
Hmm... The idea was to have "useful qualifiers" that can be reused by various parts of the code and easily edited, like ALWAYS_INLINE, IWRAM_CODE, PACKED, etc. I did notice libnds has some of them, but not all the ones that may be needed (and there is also the GBA side). In the end, it's just a personal preference though. So I can remove them.